packages/arb-token-bridge-ui/src/hooks/useTransactionHistory.ts

```diff
@@ -285,16 +304,38 @@ const useTransactionHistoryWithoutStatuses = (
     () => fetcher('withdrawals')
   )
 
-  const deposits = (depositsData || []).flat()
+  const deposits = [
+    ...getDepositsWithoutStatusesFromCache(),
+    (depositsData || []).flat()
+  ]
+
   const withdrawals = (withdrawalsData || []).flat()
 
   // merge deposits and withdrawals and sort them by date
-  const transactions = [...deposits, ...withdrawals, ...combinedCctpTransfers]
-    .flat()
-    .sort(sortByTimestampDescending)
+  const transactions = [
+    ...deposits,
+    ...withdrawals,
+    ...combinedCctpTransfers
+  ].flat()
+
+  // duplicates may occur when txs are taken from the local storage
+  const dedupedTransactions = useMemo(
+    () =>
+      Array.from(
+        new Map(
+          transactions.map(tx => [
+            `${tx.parentChainId}-${tx.childChainId}-${getTxIdFromTransaction(
```

Comment:
Dumb Q: while setting the tx-id key, have we thought about upper-lowercase considerations that might mismatch same keys while checking?

Reply:
fair point, added toLowerCase